4_chapter/dataframeImputation.jl

```diff
@@ -2,33 +2,20 @@ using Random, Statistics
 Random.seed!(0)
 include("dataframeCreation.jl")
 
-noTypePrice = (ismissing.(purchaseData[:Type])) .& 
-	(ismissing.(purchaseData[:Price]))
-purchaseData = purchaseData[.!noTypePrice,:]
-
-purchaseData[ismissing.(purchaseData[:Name]), :Name] = "notRecorded"
-purchaseData[ismissing.(purchaseData[:Time]), :Time] = "12:00 PM" 
+filter!(purchaseData) do row
```

Comment:
Unfortunately, this is probably slower than the current code, right? We need to extend the pair API to `filter!` to make it efficient.

Reply:
Agreed. That is why I have said that all the changes are only suggestions. In this case I simply find `filter!` simpler to reason about and for normal statistics the performance difference is negligible.